apps/sms/test/marionette/mocks/mock_test_storages.js

```diff
@@ -0,0 +1,57 @@
+/* global Components, Services */
+'use strict';
+
+const Cu = Components.utils;
+Cu.import('resource://gre/modules/Services.jsm');
+Cu.import('resource://gre/modules/Promise.jsm');
+
+Services.obs.addObserver(function(document) {
+  if (!document || !document.location) {
+    return;
+  }
+
+  var window = document.defaultView;
+  var storages = new Map();
+
+  function getStorageDeferred(key) {
+    var deferred = storages.get(key);
+
+    if (!deferred) {
+      deferred = {
+        data: null
+      };
+
+      deferred.promise = new Promise(function(resolve, reject) {
+        deferred.resolve = resolve;
+        deferred.reject = reject;
+      });
```

Comment:
Hmm, I didn't use it because mdn says it's obsoleted: https://developer.mozilla.org/en-US/docs/Mozilla/JavaScript_code_modules/Promise.jsm